---
title: "Poster for Evolution meeting (Providence, R. I. June 21-25, 2019)"
date: "`r format(Sys.time(), '%B %d, %Y')`"
output: 
#  pdf_document:
#  word_document
  html_document
csl: ../references/bibliography/nature.csl
bibliography: ../references/bibliography/biblio.bib
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
## Title
*Dolichospermum* evolution during during cyanobacterial bloom: insights from metatranscriptomics.

## Authors & Affiliations
Sébastien Renaut^1,2^,Nathalie Fortin^3^,Charles W. Greer^3^,B. Jesse Shapiro^2^


^1^Institut de Recherche en Biologie Végétale, Université de Montréal, Montréal, Québec, Canada,^2^Department of Biological Sciences, Université de Montréal, Montreal, Quebec, H3T 1J4, Canada,^3^Energy Mining and Environment, National Research Council Canada, Montreal, QC, Canada  


## Abstract  
Specific environmental conditions, such as elevated levels of nutrients from human activities, warmer temperatures, still water, and sunlight promote the growth of cyanobacteria, which at high density, form harmfull algae blooms. Each summer, cyanobacteria make an appearance in Lake Champlain (Baie Missisquoi, Qc) and pose health risks to humans and animals due to the production of cyanotoxins. Here, we used a metatranscriptomics approach that targets genes expressed in the water to identify changes in the bacterial community. At the height of the bloom (mid-September), chlorophyll A levels increased by nearly 100 fold and communities were dominated by the cyanobacteria *Dolichospermum circinale* and to a lesser extent *Microcystis aeruginosa*. We identified 7,000 *Dolichospermum* SNPs that showed changes in allele frequencies from the begining of the bloom to the end of the season, showing a highly dynamic evolution of the system. Nineteen genes showed the same repeatable patterns of evolution in all three sampling sites (expected: three) and these represent good candidates to predict blooms in the future.

## Introduction
Specific environmental conditions, such as elevated levels of nutrients from human activities, warmer temperatures, still water, and sunlight promote the growth of cyanobacteria, which at high density, form harmfull algae blooms. Each summer, cyanobacteria make an appearance in Lake Champlain (Baie Missisquoi) and pose health risks to humans and animals due to the production of cyanotoxins. 

Use a metatranscriptomics approach that targets genes expressed in the water to identify changes in the bacterial community during the summer. Identify *Dolichospermum* (main bloom causing cyanobacteria) genes that change in allele (SNPs) frequency from the beginning to the end of the bloom. 


## Method
###water sampling
Water sampled at three location in Lake Champlain (**see Figure 1**) during the cyanobacterial bloom in the summer of 2016 (June 1st - October 10th).

###environmental data
chlorophyll

###RNA seq preparetion
RNA extracted, followed by High Throughput sequencing (100bp paired end Illumina HiSeq).

###metatranscriptomics
Metatranscriptomes processed using SAMSA2 pipeline (trimming, merging, removal of rRNA, annotation) [@westreich2018samsa2].

###Dolicho de novo transcriptome, alignements and SNP calling
Sequences specifically annotated to *Dolichospermum* (the main cyanobacteria responsible for the bloom) used to produce a *de novo* reference transcriptome assembly using Trinity [@grabherr2011trinity]

*Dolichospermum* sequences aligned to transcriptome using bowtie2 [@langmead2012fast]. SNP called using SAMtools [@li2009sequence].

#transcriptome assessment

###regressions
Changes in allele frequency quantified using regressions (logistic regression for St1 & St2, linear for PRM given that we had few datapoints) in R

###candidate genes
Candidate genes showing significant changes (FDR corrected *p*-value < 0.05) in allele frequency matched to reference database and compared against reference transcriptome.  



## Results 


![](../results/dolicho/figures/sampling.png){width=500px}

&nbsp;
&nbsp;
&nbsp;

 **Figure 1: A map of sampling sites**
 
 ***


![](../figures/chloro.png){width=200px}

&nbsp;

**Figure 2: Chlorophyll concentration peaked in mid-September (August 1st: 7.8 ug/L, Sept 15th: 547 ug/L) resulting in a cyanobacterial bloom**  

***


&nbsp;
&nbsp;
&nbsp;

![](../results/dolicho/figures/Champlain_genera_barplot.png){width=600px}

&nbsp;

 **Figure 3: Communities dominated by Gram negative *Fusobacterium* & *Acinetobacter* prior to bloom. During Bloom, *Dolichospermum circinale* rose in frequency, followed to a lesser extent by *Microcystis aeruginosa* in September.** 
 
***

&nbsp;
&nbsp;
&nbsp;

![](../results/dolicho/figures/venn_figure.png){width=500px}
&nbsp;

**Figure 4: 72,000 SNPs identified in 2,286 *Dolichospermum* genes. A.Venn diagram: Number of SNPs (genes) showing significant changes in allele frequency from beginning (June 1st) to end of bloom (October 6th). C. Null distributions of the number of SNPs expected to show significant allele frequency change in all three sampling sites compared to observed (black vertical line, 22 SNPs observed, *p*-value < 0.0001). B. Example of a significant SNP in both St1, St2 (logistic regressions) and PRM (linear regression) sites.**

***

![](../results/dolicho/figures/go_figure.png){width=500px}



**Figure 5: A. Gene ontology groups for candidate genes (no significant over-represented groups compared to reference transcriptome). B. List of candidate genes.**


***

&nbsp;
&nbsp;
&nbsp;

## Conclusions
* Metatranscriptomics can identify cyanobacteria known to cause blooms (*Dolichospermum circinale*, *Microcystis aeruginosa*).
* ~7,000 SNPs (10% of all SNPs) show changes in allele frequencies during the bloom in at least one of the sampling site, showing a highly dynamic evolution of the system. Nineteen genes show repeatable changes in allele frequencies in all three sampling sites. These represent candidates genes whose changes in allele frequencies may help to predict blooms in the future.

* Further sampling and sequencing in progress in other lakes in order to confirm patterns observed here.

###


```{r extra, echo=FALSE, eval=F}  

## Hypotheses
# Let's make some predictions

* metag and metat alpha-diversity are well correlated.
* DNA diversity exceed RNA diversity, suggesting that pathways are not proportionally transcribed by the community species that encode them.
* Many species have cyanotoxin genes, but few dominate.
* build a model to see what influences the correlation: high vs low expression genes? alpha diversity? 

* in Dolicho, most genes alleles frequencies should show the same pattern over time. If some genes show a diff. pattern, then they are under diff. selections. In those genes are related to cyanotoxins somehow, then this is really cool. Also, they could be diff expressed? or have non-synonymous mutations?   
1.Grep all Dolicho sequences in all samples.
2.Make a reference
3.Call SNPs on a sample per sample basis
4.Look at allele freq and find outliers
5.Investigate outliers.



## Intro
There are cyano blooms (dolicho + microcystis). The are costly. We want to use genetics to understand them better


## Hypo
Metat is a good method to ID blooms
Metat and metag are highly correlated
Few cyanot genes expressed

## Results

  + 1. Metat identifies the blooms correctly (barplot + foldchanges)
    + 1.1 There is a weird sludge on Sept 15th. (associated with peak i N, P, Cl - ugly picture of bloom)
  + 2. Metat and metag are highly correlated (cor plot, but also see how metag IDs the Microcystis better).
    + 2.1 Metat and metag are not well correlated for high expression genes (high expression)
  + 3. Many cyanotoxic genes expressed (barplot cyano)
  
  #Are these patterns repeatble: we are sampling more lakes?
  #Can some genes predict the evolution of the community of the year?
  
  


 Reference to Franzosa et al. 2018 (HUMAnN2)

```

```{r stats, echo=FALSE, eval=F}  
3. Transcriptome assembly + nb SNPs : Table 1 
nb genes: 4,897
nb genes with SNP: 2,286
nb SNPs: 72,055
nb significant SNPS
St1 (logistic regression): 1,493
St2 (logistic regression): 4,249
PRM (linear regression): 3,118
St1 $\cap$ St2: 282
St1 $\cap$ PRM: 388
St2 $\cap$ PRM: 160 
St1 $\cap$ St2 $\cap$ PRM: 22 (19 genes)
```

This poster was made with ![](figures/r.jpeg){width=18px} [@r2019] and ![](figures/posterdownlogo.png){width=18px} [@posterdown]. 


## References
